pkg/kubelet/network/generic/generic.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generic
+
+import (
+	"errors"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
+	"github.com/golang/glog"
+)
+
+type genericNetworkPlugin struct {
+	host network.Host
+}
+
+const (
+	genericPluginName = "generic"
+	initCmd           = "--init"
+	preLaunchCmd      = "--pre-pod-launch"
+	preDeleteCmd      = "--pre-pod-delete"
+	executable        = "/usr/local/bin/kubelet-plugins/net/generic/generic"
```

Comment:
That should probablyu be a flag, but that sounds like a fair default.  Of
course my ubuntu machine does not even have /usr/libexec - how "modern" is
this notion?

On Tue, Mar 17, 2015 at 8:32 PM, Eric Paris <notifications@github.com>
wrote:

> In pkg/kubelet/network/generic/generic.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5069#discussion_r26636094>
> :
>
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/dockertools"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/network"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/exec"
> > +	"github.com/golang/glog"
> > +)
> > +
> > +type genericNetworkPlugin struct {
> > +	host network.Host
> > +}
> > +
> > +const (
> > +	genericPluginName = "generic"
> > +	initCmd           = "--init"
> > +	preLaunchCmd      = "--pre-pod-launch"
> > +	preDeleteCmd      = "--pre-pod-delete"
> > +	executable        = "/usr/local/bin/kubelet-plugins/net/generic/generic"
>
> You got it @mrunalp <https://github.com/mrunalp> . The "linux" way these
> days for binaries, intended to be run by other binaries and not by humans,
> is /usr/libexec/.
>
> Perfectly reasonable to bundle them in /usr/libexec/kubernetes/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5069/files#r26636094>
> .
>